### DISCUSSION TITLE
Assigning multiple detections as tags to clips

### DISCUSSION THREAD

**Discussion by PovilasID** [blakeblackshear/frigate#6907] [Ideas]
Now clips have a primary detection type: car, person, dog etc. However when multiple objects enter the field I either get one extended clip but just with a single tag though there was both a `car` and `person` detected in clip or multiple clips starting and stopping as new detections occur for example when multiple people enter and leave the frame (I had a group of people loitering around the edge of the frame). 

I am not sure how this works now exactly however I think it would be good to decouple the idea of detection and clips/recordings, so that one clip could have multiple tags for all detected objects during the timeframe it was filming. Clip could be anchored on the recording time and any detections could be assigned as tags permitting 1:many relationship. 

Also multiple snapshots for different types of detection would be nice to.

**Comment by NickM-27** [comment]
An event is essentially just a tracked object. If a person and a car enter the frame at similar times and there are not 2 events created then that's an issue with the config.

**Reply by PovilasID** [reply]
There maybe that only one of the objects detected I do not know.
However when to events created? Why? It's the same event person exited a car...
It could be one event just with multiple tags for all detected object types

**Reply by NickM-27** [reply]
An event is essentially just a tracked object, so there will be an event for every tracked object. There is cleanup logic that will delete duplicates that happen at the same time like multiple people walking in.

A car and a person really couldn't be the same event because the car event continues long after a person goes inside for example.
